test/middleware/srgssr.spec.js

```diff
@@ -41,12 +43,253 @@ describe('SrgSsr', () => {
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addBlockedSegments(player, [{}, {}]);
+
+      expect(result).toHaveLength(0);
+    });
+
+    it('Should add 2 blocked segments', async () => {
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addBlockedSegments(player, [{
+        blockReason: 'GEOBLOCK',
+        markIn: 10_0000,
+        markOut: 25_0000
+      }, {
+        markIn: 25_0000,
+        markOut: 50_0000
+      }, {
+
+        blockReason: 'ENDDATE',
+        markIn: 50_0000,
+        markOut: 60_0000
+      }, {
+        markIn: 60_0000,
+        markOut: 70_0000
+      }]);
+
+      expect(result).toHaveLength(2);
+    });
+  });
+
+  /**
+   *****************************************************************************
+   * addChapters ***************************************************************
+   *****************************************************************************
+   */
+  describe('addChapters', () => {
+    it('should not call addChapters if the chapters parameter is not an array or if the array is empty', async () => {
+      const spyOnPillarboxTextTrack = jest.spyOn(Pillarbox, 'TextTrack');
+
+      SrgSsr.addChapters(player, true);
+      SrgSsr.addChapters(player, null);
+      SrgSsr.addChapters(player, '');
+      SrgSsr.addChapters(player, undefined);
+
+      expect(spyOnPillarboxTextTrack).not.toHaveBeenCalled();
+    });
+
+    it('should remove chapters track if any', async () => {
+      const spyOnRemoveTrack = jest.spyOn(player.textTracks(), 'removeTrack');
+
+      player.textTracks().getTrackById.mockReturnValueOnce({});
+      SrgSsr.addChapters(player);
+
+      expect(spyOnRemoveTrack).toHaveBeenCalled();
+    });
+
+    it('should not add the chapter if the only available chapter is the main chapter', async () => {
+      const chapterUrn = 'urn:full:length';
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addChapters(player, chapterUrn, [{
+        fullLengthMarkIn: 0,
+        fullLengthMarkOut: 10000
+      }]);
+
+      expect(result).toHaveLength(0);
+    });
+
+    it('should add all chapters that are not the main chapter', async () => {
+      const chapterUrn = 'urn:full:length';
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addChapters(player, chapterUrn, [{
+        fullLengthMarkIn: 0,
+        fullLengthMarkOut: 10000
+      }, {
+        fullLengthUrn: chapterUrn,
+        fullLengthMarkIn: 2500,
+        fullLengthMarkOut: 5000
+      }, {
+        fullLengthUrn: chapterUrn,
+        fullLengthMarkIn: 6000,
+        fullLengthMarkOut: 9500
+      }]);
+
+      expect(result).toHaveLength(2);
```

Comment:
It could be interesting to check that the start time and end time where correctly transformed into seconds:

```suggestion
      expect(result).toHaveLength(2);
      expect(JSON.parse(result[0]).startTime).toBe(0);
      expect(JSON.parse(result[0]).endTime).toBe(1000);
```

Reply:
@jboix that's the case that should fail but I get the idea 🥲